File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
    int n = nums.size();
    long long S = (long long)n * (n + 1) / 2;
    long long SS = (long long)n * (n + 1) * (2 * n + 1) / 6; 
    long long actualSum = 0, actualSumSquares = 0;
    for (int num : nums) {
        actualSum += (long long)num;
        actualSumSquares += (long long)num * num;
    }
    long long sumDiff = actualSum - S;  // x - y
    long long squareSumDiff = actualSumSquares - SS;  // x^2 - y^2
    long long sumXY = squareSumDiff / sumDiff;  // x + y
    int x = (sumDiff + sumXY) / 2;
    int y = sumXY - x;
    return {x, y};       
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
         HashSet<Integer> hs=new HashSet<>();
   int duplicate=-1;
   for(int i=0;i<nums.length;i++){
     
       if(!hs.contains(nums[i])){
           hs.add(nums[i]);
       }else{
           duplicate =nums[i];
       }
   }

int missing=-1;
int n=nums.length;
for(int i=0;i<=n;i++){
    if(!hs.contains(i)){
        missing =i;
    }
}
int ans[]={duplicate,missing};
return ans;
    }
}